### DISCUSSION TITLE
"setTimeout" Function migration from J2V8

### DISCUSSION THREAD

**Discussion by darekbx** [caoccao/Javet#414] [Q&A]
Hi, we are migrating our project from J2V8 to Javet and we love it's simplity when using `bind` method. However, we've found a problem with implementation of `setTimeout` function, when using pure V8 and JS React. 

Here's how is implemented `setTimeout` (simplified code):
```kotlin
class SchedulerRuntime {
  @V8Function
  fun setTimeout(function: V8ValueFunction, timeout: V8ValueInteger) {
      Timer().schedule(timerTask {
          function.callVoid(null, null)
      }, timeout.value.toLong())
  }
  //...
}
//...
v8Runtime.globalObject.bind(SchedulerRuntime())
```

When we are calling this method we see this exception:
```
com.caoccao.javet.exceptions.JavetException: Runtime is already closed
at com.caoccao.javet.values.reference.V8ValueReference.checkV8Runtime(V8ValueReference.java:45)
at com.caoccao.javet.values.reference.V8ValueFunction.callExtended(V8ValueFunction.java:97)
at com.caoccao.javet.values.reference.IV8ValueFunction.callVoid(IV8ValueFunction.java:305)
``` 

Part of the compiled React JS code:
```javascript
//...
  scheduleTimeout: function (e, t) {
    return setTimeout(() => {
      try {
        e();
      } catch (e) {
        console.error(`${e}`);
      }
    }, t || 0);
  },
//...
```

Without `Timer()` function is being called properly, but when is called from a thread then `V8ValueFunction.isClosed` flag is set to `true`. 

PS. I was also trying to add `async` keword for `setTimeout` function, according to [testAsyncFunction](https://github.com/caoccao/Javet/blob/main/src/test/java/com/caoccao/javet/values/reference/TestV8ValueFunction.java) test, but still is not working and flag `V8ValueFunction.isAsyncFunction` was set to `false`.

**Comment by caoccao** [comment]
There are generally 3 options for you to choose.

1. Use the Node.js mode. The Node.js mode has everything you need.
2. Reference my another project [Javenode](https://github.com/caoccao/Javenode) which implements a set of such API.
3. Borrow the implementation of `setTimeout()` from [Javenode](https://github.com/caoccao/Javenode) in your project.

**Comment by darekbx** [comment]
@caoccao Thanks! I'll borrow implementation from Javenode.
